clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -668,7 +669,9 @@ void SYCL::fpga::BackendCompiler::ConstructJob(
     const InputInfoList &Inputs, const ArgList &Args,
     const char *LinkingOutput) const {
   assert((getToolChain().getTriple().getArch() == llvm::Triple::spir ||
```

Comment:
Please, use .isSPIR() and .isSPIRV() methods.

Reply:
Yes. That's handy. Thanks @bader. Change in upcoming commit.